app/src/main/java/eu/theinvaded/mastondroid/ui/activity/LoginActivity.java

```diff
@@ -29,20 +44,37 @@ protected void onCreate(Bundle savedInstanceState) {
         this.context = this;
         preferences =
                 context.getSharedPreferences(getString(R.string.preferences), context.MODE_PRIVATE);
-
         binding = DataBindingUtil.setContentView(this, R.layout.activity_login);
-        LoginViewModel loginViewModel = new LoginViewModel(this);
+        loginViewModel = new LoginViewModel(this);
         binding.setViewModel(loginViewModel);
+
+        Uri uri = getIntent().getData();
+        treatAuthorization(uri);
+
+        if (preferences.contains(getString(R.string.authKey))) {
+            startMainActivity();
+        }
     }
 
-    @Override
-    public String getUsername() {
-        return binding.usernameTv.getText().toString();
+    private void treatAuthorization(Uri uri) {
+        if (uri != null ){
```

Comment:
This and others like that, not a great style, due to increased complexity. Please try to cancel early like this:

`if (uri != null ) return;`

That code is also not tidied. I don't know what IDE you use. Android Studio comes with very strong development tools and a keyboard shortcut ctrl+alt+L to reformat code according to common rules.